### DISCUSSION TITLE
Migration up fail

### DISCUSSION THREAD

**Discussion by nishtjak-github** [mikro-orm/mikro-orm#6048] [General]
### Describe the bug

I got this entity:

```
@Entity()
export class PetrolStation {

	@PrimaryKey()
	id!: number;

	@Property({ type: "varchar" })
	siteId!: string;

	@Enum(() => StationBrand)
	brand!: StationBrand;

	@Property({ type: "text"})
	address!: string;

	@Property({ type: "varchar" })
	postcode!: string;

	@Property( {onCreate: () => new Date() } )
	createdAt = new Date();

	@Property({ onUpdate: () => new Date() })
	updatedAt = new Date();


	constructor(siteId: string, brand: StationBrand, address: string, postcode: string, createdAt: Date) {
		this.siteId = siteId;
		this.brand = brand;
		this.address = address;
		this.postcode = postcode;
		this.createdAt = createdAt;
	}
}

export enum StationBrand {
	APPLEGREEN = "APPLEGREEN",
	BP = "BP",
	SHELL = "SHELL",
	JET = "JET",
	TEXACO = "TEXACO",
	ASDA = "ASDA",
	ESSO = "ESSO",
	GULF = "GULF",
	HARVESTENERGY = "HARVESTENERGY",
	MURCO = "MURCO",
	MORRISONS = "MORRISONS",
	SAINSBURYS = "SAINSBURYS",
	TESCO = "TESCO"
}
```

Running the initial migration create cli command produces the following migration file:
```
import { Migration } from '@mikro-orm/migrations';

export class Migration20240919200054 extends Migration {

  override async up(): Promise<void> {
    this.addSql(`create table "petrol_station" ("id" serial primary key, "site_id" varchar(255) not null, "brand" text check ("brand" in ('APPLEGREEN', 'BP', 'SHELL', 'JET', 'TEXACO', 'ASDA', 'ESSO', 'GULF', 'HARVESTENERGY', 'MURCO', 'MORRISONS', 'SAINSBURYS', 'TESCO')) not null, "address" text not null, "postcode" varchar(255) not null, "created_at" varchar(255) not null, "updated_at" timestamptz not null);`);
  }

}

```

Then, running migration:up gives the following error:
```
MigrationError: Migration Migration20240919200054 (up) failed: Original error: create table "petrol_station" ("id" serial primary key, "site_id" varchar(255) not null, "brand" text check ("brand" in ('APPLEGREEN', 'BP', 'SHELL', 'JET', 'TEXACO', 'ASDA', 'ESSO', 'GULF', 'HARVESTENERGY', 'MURCO', 'MORRISONS', 'SAINSBURYS', 'TESCO')) not null, "address" text not null, "postcode" varchar(255) not null, "created_at" varchar(255) not null, "updated_at" timestamptz not null); - relation "petrol_station" already exists
    at /app/node_modules/umzug/src/umzug.ts:259:17
    at processTicksAndRejections (node:internal/process/task_queues:105:5)
    at async Umzug.runCommand (/app/node_modules/umzug/src/umzug.ts:208:14)
    at async Migrator.runInTransaction (/app/node_modules/@mikro-orm/migrations/Migrator.js:313:21)

```
 ```
jse_cause: TableExistsException: create table "petrol_station" ("id" serial primary key, "site_id" varchar(255) not null, "brand" text check ("brand" in ('APPLEGREEN', 'BP', 'SHELL', 'JET', 'TEXACO', 'ASDA', 'ESSO', 'GULF', 'HARVESTENERGY', 'MURCO', 'MORRISONS', 'SAINSBURYS', 'TESCO')) not null, "address" text not null, "postcode" varchar(255) not null, "created_at" varchar(255) not null, "updated_at" timestamptz not null); - relation "petrol_station" already exists
      at PostgreSqlExceptionConverter.convertException (/app/node_modules/@mikro-orm/postgresql/PostgreSqlExceptionConverter.js:46:24)
      at PostgreSqlDriver.convertException (/app/node_modules/@mikro-orm/core/drivers/DatabaseDriver.js:345:54)
      at /app/node_modules/@mikro-orm/core/drivers/DatabaseDriver.js:349:24

```

I am not sure what else to do, I have checked that I can run the generated SQL myself and it does create the table if I run the up migration via DataGrip.
Is there something I am missing?

Edit: This is a brand new postgres db on Aiven service, maybe its something to do with that. However the migration table gets created without problems

### Reproduction

```

@Entity()
export class PetrolStation {

	@PrimaryKey()
	id!: number;

	@Property({ type: "varchar" })
	siteId!: string;

	@Enum(() => StationBrand)
	brand!: StationBrand;

	@Property({ type: "text"})
	address!: string;

	@Property({ type: "varchar" })
	postcode!: string;

	/*@Property({ type: GeographyType, nullable: true })
	location: Point;
	*/
	@Property( {onCreate: () => new Date() } )
	createdAt = new Date();

	@Property({ onUpdate: () => new Date() })
	updatedAt = new Date();

	/*@OneToMany("FuelPrice", (fuelPrice: FuelPrice) => fuelPrice.station)
	fuelPrices = new Collection<FuelPrice>(this);*/


	constructor(siteId: string, brand: StationBrand, address: string, postcode: string, location: Point, createdAt: Date) {
		this.siteId = siteId;
		this.brand = brand;
		this.address = address;
		this.postcode = postcode;
		//this.location = location;
		this.createdAt = createdAt;
	}
}

export enum StationBrand {
	APPLEGREEN = "APPLEGREEN",
	BP = "BP",
	SHELL = "SHELL",
	JET = "JET",
	TEXACO = "TEXACO",
	ASDA = "ASDA",
	ESSO = "ESSO",
	GULF = "GULF",
	HARVESTENERGY = "HARVESTENERGY",
	MURCO = "MURCO",
	MORRISONS = "MORRISONS",
	SAINSBURYS = "SAINSBURYS",
	TESCO = "TESCO"
}

```

Run: `npx mikro-orm migration:create --initial`
Run: `npx mikro-orm migration:up`

### What driver are you using?

@mikro-orm/postgresql

### MikroORM version

6.3.10

### Node.js version

node: 22.8.0, typescript: 5.6.2

### Operating system

macos 14.3.1

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/mikro-orm/mikro-orm/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://mikro-orm.io/docs).
- [X] Check that there isn't [already an issue](https://github.com/mikro-orm/mikro-orm/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/mikro-orm/mikro-orm/discussions) or join our [Discord](https://discord.gg/w8bjxFHS7X).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by B4nan** [comment]
Are you sure there is just one migration calling this `create table` query? I can imagine you somehow created two, and since they are running inside a master transaction, one passes, and the other fails and reverts the first one.

**Reply by nishtjak-github** [reply]
That's the only migration I'm trying to run. That's the only SQL in the migrations folder. It's weird that I can run this from the console but not when I run the migration:up cli command.

You can see from the repro steps I have other fields on the entity but I'm trying to narrow down the actual problem, so I just have simple fields but no luck still

**Reply by B4nan** [reply]
Maybe its some permission issue? You said you use some service, does it work locally?

**Reply by nishtjak-github** [reply]
I mean it could be, however the migration table gets created without any problems first. I'll try postgres on my aws account also, maybe it is a permissions issue

**Comment by nishtjak-github** [comment]
Right, I have actually got an answer to this.
I have a `tsc -w` running while developing. Since I havent excluded the migrations folder from being transpiled we get a js file out of it. Thus running `npx mikro-orm migration:up` runs the .ts file and the .js, which contains the same migration and in turn causing the error. Running just the .ts or .js migration, I get no error. IntelliJ doesnt help that it hides the js files when you developing with typescript.

@B4nan Your initial hunch was correct, there was an extra migration, the same one! I just had to sleep on it to realise whats going on. 
Thanks for your help.
